runtime/limits_integ_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package main
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type commandResult struct {
+	stdout   string
+	stderr   string
+	exitCode uint32
+}
+
+func runTask(ctx context.Context, c containerd.Container) (*commandResult, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	task, err := c.NewTask(ctx, cio.NewCreator(cio.WithStreams(nil, &stdout, &stderr)))
+	if err != nil {
+		return nil, err
+	}
+
+	exitCh, err := task.Wait(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = task.Start(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case exitStatus := <-exitCh:
+		if err := exitStatus.Error(); err != nil {
+			return nil, err
+		}
+
+		_, err := task.Delete(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		return &commandResult{
+			stdout:   stdout.String(),
+			stderr:   stderr.String(),
+			exitCode: exitStatus.ExitCode(),
+		}, nil
+	case <-ctx.Done():
+		return nil, errors.New("context cancelled")
+	}
+}
+
+func TestDiskLimit_Isolated(t *testing.T) {
```

Comment:
nit: I'd move `TestDiskLimit_Isolated` up to the top of the file since this is actually the entry point and the other functions are helpers.

Reply:
Okay. Will do.

Reply:
Let me move runTask to integ_test.go since it can be used from other tests.